### PR TITLE
Nginx

### PR DESCRIPTION
**Pull request by beardyunixer**
This was...a bit out of date.

### CODE REVIEW

File: mods/sample-nginx.config [hunk 1]
```diff
@@ -52,24 +52,40 @@ server {
...
+  #include standard.conf
   # allow uploads up to 20MB in size
   client_max_body_size 20m;
   client_body_buffer_size 128k;
 
   # rewrite to front controller as default rule
   location / {
-    rewrite ^/(.*) /index.php?q=$uri&$args last;
+    if ($is_args != "") {
+        rewrite ^/(.*) /index.php?q=$uri&$args last;
```
Comment (annando) [inline mods/sample-nginx.config]: We are now using the variable ```pagename``` like you can see here: https://github.com/friendica/friendica/blob/develop/.htaccess#L38

File: mods/sample-nginx.config [hunk 2]
```diff
@@ -52,24 +52,40 @@ server {
...
+    }
+    rewrite ^/(.*) /index.php?q=$uri last;
```
Comment (annando) [inline mods/sample-nginx.config]: Same like above.

File: mods/sample-nginx.config [hunk 3]
```diff
@@ -122,5 +138,7 @@ server {
   location ~ /\. {
     deny all;
   }
+
```
Comment (annando) [inline mods/sample-nginx.config]: Why is here an empty line?

**Comment by anmol26s** [comment]
I am not able forcefully redirect to https. Can you please suggest changes to the nginx configuration?
https://github.com/YunoHost-Apps/friendica_ynh/blob/master/conf/nginx.conf

The line `rewrite ^ https://$server_name$request_uri? permanent; 
` should be forcing https by default,isn't it?

**Comment by beardyunixer** [comment]
You don't seem to have a server_name directive anywhere.  Presumably, you're using _ as a catch all somewhere in a default config?

In that case, you would need to use the $host variable, rather than the $server_name variable.

**Comment by anmol26s** [comment]
Surprisingly, there is server_name  in the main configuration file 
https://github.com/YunoHost/yunohost/blob/unstable/data/templates/nginx/server.tpl.conf which further points to `include conf.d/{{ domain }}.d/*.conf;` where the Friendica nginx.conf files resides having the configuration provided in the previous comment. 
It should not take the server_name from the default config?

Manually adding the domain name(eg.` rewrite ^ https://pod.domain.tld$request_uri? permanent;`) instead of `$server_name` works but it would be good if the default could work.

**Comment by beardyunixer** [comment]
I'm sorry, I don't see anything immediately wrong.

All I can suggest for now is to try replacing rewrite ^ https://$server_name$request_uri? permanent; with rewrite ^ https://$host$request_uri? permanent; or rewrite ^ https://$http_host$request_uri? permanent;

I don't think this will work.  If it does, it means $server_name is undefined - which I don't think it will be now I've seen the server.tpl.conf.  If it doesn't work, it means I can't even give you a hint until I take a closer look.

Unfortunately, I'm busy with personal problems (hence the long wait for this reply), and I don't think any other Nginx people are likely to be looking here.  I'm happy to help if you have the patience to wait, but you might find faster help at the Friendica Support Forum.

I'll have a play with YunoHost it in a VM and see if I can figure it out when I get a chance - but it will be at least a few days before I can.

**Comment by anmol26s** [comment]
I will wait for you to test as both Friendica and Hubzilla are having the same problem. My previous comment suggesting, of using domain name for server_name was not successful when I again tested it.
So this issue still remain unsolved. The weird thing is that this happen only the first time the hub is opened in the browser. On refreshing the hub it redirects to https.

I have tested both your suggestion ($host and $http_host) which makes nginx fail.
You are welcome to experiment with YunoHost.

**Comment by anmol26s** [comment]
Maybe I should have used rewrite ^ https://$server_name$uri? permanent; instead of rewrite ^ https://$server_name$request_uri? permanent;
But $uri instead of  $request_uri is a security risk see here:
https://github.com/friendica/friendica/issues/4146

**Pull request merged**